librz/core/yank.c

```diff
@@ -193,267 +163,188 @@ RZ_API int rz_core_yank_paste(RzCore *core, ut64 addr, int len) {
+RZ_API bool rz_core_yank_cat(RzCore *core, ut64 pos) {
 	int ybl = rz_buf_size(core->yank_buf);
-	if (ybl > 0) {
-		if (pos < ybl) {
-			ut64 sz = ybl - pos;
-			char *buf = RZ_NEWS(char, sz);
-			if (!buf) {
-				return false;
-			}
-			rz_buf_read_at(core->yank_buf, pos, (ut8 *)buf, sz);
-			int len = rz_str_nlen(buf, sz);
-			rz_cons_memcat(buf, len);
-			rz_cons_newline();
-			return true;
-		}
-		eprintf("Position exceeds buffer length.\n");
-	} else {
-		rz_cons_newline();
+	if (ybl <= 0) {
+		RZ_LOG_ERROR("No buffer yanked already\n");
+		return false;
 	}
-	return false;
+	if (pos >= ybl) {
+		RZ_LOG_ERROR("Position exceeds buffer length.\n");
+		return false;
+	}
+	ut64 sz = ybl - pos;
+	char *buf = RZ_NEWS(char, sz);
+	if (!buf) {
+		return false;
+	}
+	rz_buf_read_at(core->yank_buf, pos, (ut8 *)buf, sz);
+	rz_cons_memcat(buf, sz);
+	rz_cons_newline();
+	return true;
 }
 
-RZ_API int rz_core_yank_hud_file(RzCore *core, const char *input) {
-	char *buf = NULL;
-	bool res = false;
-	ut32 len = 0;
-	if (!input || !*input) {
+RZ_API bool rz_core_yank_cat_string(RzCore *core, ut64 pos) {
+	int ybl = rz_buf_size(core->yank_buf);
+	if (ybl <= 0) {
+		RZ_LOG_ERROR("No buffer yanked already\n");
+		return false;
+	}
+	if (pos >= ybl) {
+		RZ_LOG_ERROR("Position exceeds buffer length.\n");
+		return false;
+	}
+	ut64 sz = ybl - pos;
+	char *buf = RZ_NEWS(char, sz);
+	if (!buf) {
+		return false;
+	}
+	rz_buf_read_at(core->yank_buf, pos, (ut8 *)buf, sz);
+	int len = rz_str_nlen(buf, sz);
+	rz_cons_memcat(buf, len);
+	rz_cons_newline();
+	return true;
+}
+
+RZ_API bool rz_core_yank_hud_file(RzCore *core, const char *input) {
+	if (RZ_STR_ISEMPTY(input)) {
 		return false;
 	}
 	for (input++; *input == ' '; input++) {
 		/* nothing */
 	}
-	buf = rz_cons_hud_file(input);
-	len = buf ? strlen((const char *)buf) + 1 : 0;
-	res = rz_core_yank_set_str(core, RZ_CORE_FOREIGN_ADDR, buf, len);
+	char *buf = rz_cons_hud_file(input);
+	ut64 len = buf ? strlen((const char *)buf) + 1 : 0;
+	bool res = rz_core_yank_set_str(core, RZ_CORE_FOREIGN_ADDR, buf, len);
 	free(buf);
 	return res;
 }
 
-RZ_API int rz_core_yank_hud_path(RzCore *core, const char *input, int dir) {
-	char *buf = NULL;
-	ut32 len = 0;
-	int res;
+RZ_API bool rz_core_yank_hud_path(RzCore *core, const char *input, int dir) {
+	if (RZ_STR_ISEMPTY(input)) {
+		return false;
+	}
 	for (input++; *input == ' '; input++) {
 		/* nothing */
 	}
-	buf = rz_cons_hud_path(input, dir);
-	len = buf ? strlen((const char *)buf) + 1 : 0;
-	res = rz_core_yank_set_str(core, RZ_CORE_FOREIGN_ADDR, buf, len);
+	char *buf = rz_cons_hud_path(input, dir);
+	ut64 len = buf ? strlen((const char *)buf) + 1 : 0;
+	bool res = rz_core_yank_set_str(core, RZ_CORE_FOREIGN_ADDR, buf, len);
 	free(buf);
 	return res;
 }
 
-RZ_API bool rz_core_yank_hexpair(RzCore *core, const char *input) {
-	if (!input || !*input) {
+RZ_API bool rz_core_yank_hexpair(RzCore *core, const char *str) {
```

Comment:
Maybe it's just me, but I have an hard time understanding the yank functions where they mean from clipboard to file or from file to clipboard... Could you add only the `brief` doxygen doc for some of these API? even a very short one is fine.